test/governance/utils/Votes.behavior.js

```diff
@@ -86,16 +73,13 @@ function shouldBehaveLikeVotes() {
       });
 
       it('rejects reused signature', async function () {
-        const { v, r, s } = fromRpcSig(
-          ethSigUtil.signTypedMessage(
-            delegator.getPrivateKey(),
-            buildData(this.chainId, this.votes.address, this.name, {
-              delegatee: delegatorAddress,
-              nonce,
-              expiry: MAX_UINT256,
-            }),
-          ),
-        );
+        const { v, r, s } = await buildData(this.votes, {
+          delegatee: delegatorAddress,
+          nonce,
+          expiry: MAX_UINT256,
+        })
+          .then(data => ethSigUtil.signTypedMessage(delegator.getPrivateKey(), { data }))
+          .then(fromRpcSig);
```

Comment:
Can we agree to avoid `.then` except for one-liners? :sweat_smile: Let's please come to an agreement so it doesn't show up in all reviews.

Reply:
this pattern (buildData → signTypedMessage → fromRpcSig) is present many times. I fealt that having it as one block, with no extra intermediary declaration would be cleaner.

I'll probably rewrite it as "buildAndSignData"